lib/Index/IndexRecord.cpp

```diff
@@ -506,9 +506,13 @@ emitDataForSwiftSerializedModule(ModuleDecl *module,
   if (ModuleDecl *declaring = module->getDeclaringModuleIfCrossImportOverlay())
     moduleName = declaring->getNameStr().str();
 
+  // Run an initial up to date check before reloading the module. This will
+  // catch both already-reloaded modules + modules that weren't able to be
```

Comment:
This turns out not to be true and I need to add `-Rmodule-loading` to the test. The `ModuleDecl` here came from the dependencies of the adjacent-loaded module. So it's always going to be the module prior to reloading. That means that we're still always reloading here. It's still less than without this change as we'd cut off the chain early, but it's still *way* more than it needs to be.

We're also paying the cost of a single reload for every single compilation, regardless of whether we've already output the unit or not.

Rather than this change, I think I'll likely just use the original filename for the unit rather than the reloaded filename.

Reply:
... But that doesn't work because the second run will load the cached module instead.

Reply:
I've updated this to now record the interface path on load and pass that all the way through. We can now reference the interface path, even when the module was itself loaded from the adjacent swiftmodule.